rs/xnet/endpoint/src/lib.rs

```diff
@@ -282,16 +280,20 @@ impl XNetEndpoint {
         certified_stream_store: Arc<dyn CertifiedStreamStore>,
         tls: Arc<dyn TlsConfig + Send + Sync>,
         registry_client: Arc<dyn RegistryClient + Send + Sync>,
-        config: XNetEndpointConfig,
+        config: Config,
```

Comment:
Clueless question: where does this local config come from? And is it possible for it to diverge from the registry contents? (As in, registry says port _X_ but local config says port _Y_.)

Reply:
> I don't want to belabor the point, but what's to stop a replica from starting up with its XNet endpoint listening on :123 but the registry leading other replicas to try to connect to :456?

Nothing, but then the registry value should just be updated.

This said, using the registry as source of truth is more problematic because we need to dynamically stop and start the xnet endpoint. It is much simpler to know the port on startup for the server-end.


> I mean apart from the port being practically hardcoded to :2197. Because if that's all we're relying on, we may as well drop it from both the config and the registry.

The config is more or less same as the source code. But yes, ideally we should drop it from the registry given we have pocket IC that simulates xnet.

Reply:
All else being equal, I would prefer it for nodes / replicas to follow what the registry tells them to do (i.e., among other things, which port they should listen on).

But it's only my preference, so feel free to go ahead with this.